StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/SA1629DocumentationTextMustEndWithAPeriod.cs

```diff
@@ -32,28 +38,95 @@ namespace StyleCop.Analyzers.DocumentationRules
             ImmutableArray.Create(Descriptor);
 
         /// <inheritdoc/>
-        public override void Initialize(AnalysisContext context)
+        protected override void HandleXmlElement(SyntaxNodeAnalysisContext context, bool needsComment, IEnumerable<XmlNodeSyntax> syntaxList, params Location[] diagnosticLocations)
+        {
+            foreach (var xmlElement in syntaxList.OfType<XmlElementSyntax>())
+            {
+                var elementDone = false;
+                for (var i = xmlElement.Content.Count - 1; !elementDone && (i >= 0); i--)
+                {
+                    var contentNode = xmlElement.Content[i] as XmlTextSyntax;
+                    if (contentNode != null)
+                    {
+                        for (var j = contentNode.TextTokens.Count - 1; !elementDone && (j >= 0); j--)
+                        {
+                            var textToken = contentNode.TextTokens[j];
+                            var textWithoutTrailingWhitespace = textToken.Text.TrimEnd(' ', '\r', '\n');
+
+                            if (!string.IsNullOrEmpty(textWithoutTrailingWhitespace))
+                            {
+                                if (!textWithoutTrailingWhitespace.EndsWith(".", StringComparison.Ordinal))
+                                {
+                                    var location = Location.Create(xmlElement.SyntaxTree, new TextSpan(textToken.SpanStart + textWithoutTrailingWhitespace.Length, 1));
+                                    context.ReportDiagnostic(Diagnostic.Create(Descriptor, location));
+                                }
+
+                                elementDone = true;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        /// <inheritdoc/>
+        protected override void HandleCompleteDocumentation(SyntaxNodeAnalysisContext context, bool needsComment, XElement completeDocumentation, params Location[] diagnosticLocations)
         {
-            // TODO: Implement analysis
+            try
+            {
+                foreach (var node in completeDocumentation.Nodes().Cast<XElement>())
+                {
+                    var textWithoutTrailingWhitespace = node.Value.TrimEnd(' ', '\r', '\n');
+                    if (!string.IsNullOrEmpty(textWithoutTrailingWhitespace))
+                    {
+                        if (!textWithoutTrailingWhitespace.EndsWith(".", StringComparison.Ordinal))
+                        {
+                            context.ReportDiagnostic(Diagnostic.Create(Descriptor, diagnosticLocations[0], NoCodeFixProperties));
+
+                            // only report a single instance of the diagnostic, as they will all be reported on the same location anyway.
+                            break;
+                        }
+                    }
+                }
+            }
+            catch (InvalidCastException)
```

Comment:
Nothing, seems that something was left over from debugging. I'll remove it.

In more detail, I made a mistake with the included documentation, resulting in a generated xml comment from roslyn stating that there was something wrong with the included documentation. This results in an InvalidCastException on line 112.

Typing this has made me realize that if the end user makes a mistake in the included documentation then the same scenario can happen 😃 I'll adapt line 112 to use `OfType`, instead of `Cast`.